### PR TITLE
FPO-245 Review content and add validation error to new key form

### PR DESCRIPTION
**Pull request by mkayad**
### Jira link
[FPO-245](https://transformuk.atlassian.net/browse/FPO-245)
### What?

I have added/removed/altered:

- [x] Added validation error to new key form
- [x] Removed
- [x] Altered: Reviewed some of the screens content

### Why?

I am doing this because:

-

### AC

- Tested
- Shared with stakeholders
- Pair reviewed

### CODE REVIEW

File: views/keySuccessPage.njk [hunk 1]
```diff
@@ -4,31 +4,38 @@
...
+  <h2 class="govuk-heading-l">What happens next</h2>
+  <p class="govuk-body">Use the API key to access the FPO Commodity Code Identification Tool by including it in the <strong>X-Api-Key</strong> header of any calls you make.</p>
+  {{ govukWarningText({
+    text: "You must copy this key to somewhere safe. You will not be able to view this API key again once you leave this page.",
+    iconFallbackText: "Warning"
+  }) }}
+
+  <p class="govuk-body">
+  {{ govukInsetText({ text: secretHtml|safe }) }}
+  <button class="govuk-button govuk-button--secondary" data-module="govuk-button" onclick="copyToClipboard()">Copy to clipboard</button>
```
Comment (willfish) [inline views/keySuccessPage.njk:24]: I think we ought to be able to just use the attributes field and get the benefits of using the nunjucks macro:

```nunjucks
{{ govukButton({
  text: "Copy to clipboard",
  classes: "govuk-button--secondary",
  attributes: {"onclick": "copyToClipboard()"}
}) }}
```

File: views/keySuccessPage.njk [hunk 2]
```diff
@@ -4,31 +4,38 @@
...
+  </p>
+
+  <p class="govuk-body">
+      <a class="govuk-link" href="/dashboard/{{ organisationId }}">Back to dashboard</a>
```
Comment (willfish) [inline views/keySuccessPage.njk:28]:
```suggestion
      <a class="govuk-link" href="/dashboard">Back to dashboard</a>
```

File: src/routes/dashboardRoutes.ts [hunk 3]
```diff
@@ -18,7 +19,7 @@ if (isProduction) { router.use(requiresAuth()) }
 router.get('/', showDashboard)
 
 router.get('/new', newKey)
-router.post('/create', create)
+router.post('/create', body('description', 'Enter the description for your API key').notEmpty(), create)
```
Comment (willfish) [inline src/routes/dashboardRoutes.ts:22]: I think you don't need this here?

Comment (willfish) [reply src/routes/dashboardRoutes.ts:22]:
```suggestion
router.post('/create', body('description', create)
```

File: views/keySuccessPage.njk (hunk 2 above)
Comment (mkayad) [reply views/keySuccessPage.njk:28]: will fix, I think it is a rebase / merge issue

File: views/keySuccessPage.njk (hunk 1 above)
Comment (mkayad) [reply views/keySuccessPage.njk:24]: make sense, will replace it

File: src/routes/dashboardRoutes.ts (hunk 3 above)
Comment (mkayad) [reply src/routes/dashboardRoutes.ts:22]: I thought that is how the express-validator works, if this is not the place, where else is more appropriate?

Comment (willfish) [reply src/routes/dashboardRoutes.ts]: I figured you were basically injecting content into the context for the rendered template, here, that's actually happening elsewhere? If we remove it my guess was that we still have the error in the context of the template

Comment (willfish) [reply src/routes/dashboardRoutes.ts]: I could be wrong but just noticed a bit of duplication with this message

**Review by willfish — APPROVED** [review]

**Review by willfish — APPROVED** [review]

**Review by Lee365 — APPROVED** [review]
LGTM

**Pull request merged**